rfcs/test_scoring_change.md

```diff
@@ -0,0 +1,92 @@
+# RFC: Change results aggregation and visualization on wpt.fyi to increase accuracy and utility
+
+## Summary
+
+The aggregate scoring that is displayed in the results view of wpt.fyi has
+consistently been inaccurate in some aspects. This proposal will describe a
+change to the way that test results are aggregated, as well as a change to
+how those results are displayed.
+
+## Details
+There are a number of aspects to this change. Each subsection will describe
+these aspects in detail.
+### Remove harness status from subtest aggregation
+
+When interpreting the results of test suite runs, a status is present that
+represents either the overall test's status or the "harness status", which
+represents whether any harness errors were present for `testharness.js` test
+runs. The current results aggregation method counts this harness status as a
+separate subtest itself, which can inflate the overall passing percentage of
+a test. There are many instances of a test outright failing its only subtest,
+but the test is weighted as 50% passing because there were no harness errors
+[(example)](https://wpt.fyi/results/audio-output/selectAudioOutput-sans-user-activation.https.html?run_id=5642791446118400&run_id=5668568732532736&run_id=5739999172493312&run_id=5735075831349248).
+The solution proposed is to no longer aggregate this harness status into the
+subtest totals.
+
+### Count test as a failure if a harness error exists
+
+Although it seems intuitive to simply disregard the harness status when
+calculating tests, there are instances where a harness error will occur, which
+causes a number of subtests to not be executed. In some circumstances, it is
+possible that a harness error will occur, allowing for some subtests to run
+and pass as expected, but some subtests to be not run and not be registered,
+as they are not present during aggregation. This has the possibility of hiding
+the scope of failure for a specific test in this scenario. The solution
+proposed, although not perfect, is to mark these tests with harness errors
+as 0% passing, regardless of subtest results.
+
+_Note: Individual subtest results will still be visible normally even with_
```

Comment:
I think the caveat here is important: this change doesn't seem to help browser developers at all; they still need to go down into the individual test and figure out "oh actually there are some passing subtests, the problem is that the test is slow running, so we time out" (or whatever).

Reply:
It is true that this change isn't perfect, but hopefully still a net positive.

To use the example you mentioned, it could be the case that, for example, 7 subtests pass as expected, but a timeout occurs on the 8th test. There are instances where a test like this will be marked as 7/8 subtests passing (or 7/9 since the harness status is currently counted as a subtest), but the test could have many more subtests that were not run and are missing from the results. The current aggregation method can hide the scope of an issue like this, so the hope is that this change will bring attention to areas of improvement that could currently be overlooked.

That being said, I'm not certain about how common this scenario is. Perhaps delving into this further and having numbers would be of use.